ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/StreamedQueryResourceTest.java

```diff
@@ -131,6 +141,58 @@ public void shouldReturn400OnBadStatement() throws Exception {
     assertThat(errorMessage.getMessage(), containsString("some error message"));
   }
 
+  @Test
+  public void shouldNotWaitIfCommandSequenceNumberSpecified() throws Exception {
+    // Given:
+    replay(replayableCommandQueue);
+
+    // When:
+    testResource.streamQuery(new KsqlRequest(queryString, Collections.emptyMap(), null));
+
+    // Then:
+    verify(replayableCommandQueue);
+  }
+
+  @Test
+  public void shouldWaitIfCommandSequenceNumberSpecified() throws Exception {
+    // Given:
+    replayableCommandQueue.ensureConsumedUpThrough(anyLong(), anyLong());
+    expectLastCall();
+
+    replay(replayableCommandQueue);
+
+    // When:
+    testResource.streamQuery(new KsqlRequest(queryString, Collections.emptyMap(), 3L));
+
+    // Then:
+    verify(replayableCommandQueue);
+  }
+
+  @Test
+  public void shouldReturn503IfTimeoutWhileWaitingForCommandSequenceNumber() throws Exception {
+    // Given:
+    replayableCommandQueue.ensureConsumedUpThrough(anyLong(), anyLong());
+    expectLastCall().andThrow(new TimeoutException("whoops"));
+
+    replay(replayableCommandQueue);
+
+    try {
+      // When:
+      testResource.streamQuery(new KsqlRequest(queryString, Collections.emptyMap(), 3L));
+
+      // Then:
+      fail("Should throw KsqlRestException in response to timeout.");
+    } catch (KsqlRestException e) {
+      final Response response = e.getResponse();
+      assertThat(response.getStatus(), is(Response.Status.SERVICE_UNAVAILABLE.getStatusCode()));
```

Comment:
FYI, using the same custom Mathers stuff I mentioned above, it's possible to do:

```java
@Test
  public void shouldReturn503IfTimeoutWhileWaitingForCommandSequenceNumber() throws Exception {
    // Given:
    replayableCommandQueue.ensureConsumedUpThrough(anyLong(), anyLong());
    expectLastCall().andThrow(new TimeoutException("whoops"));

    replay(replayableCommandQueue);
    
    expectedException.expect(KsqlRestException.class);
    expectedException.expect(statusCode(Response.Status.SERVICE_UNAVAILABLE));
    expectedException.expect(errorCode(Errors.ERROR_CODE_COMMAND_QUEUE_CATCHUP_TIMEOUT));
    expectedException.expect(errorMessage("whoops"));

   // When:
   testResource.streamQuery(new KsqlRequest(queryString, Collections.emptyMap(), 3L));
  }
```

Because `expect` can take a `Matcher`
Which is pretty neat. Not that I'm saying you should. Just saying you can. Up to you.